src/server/game/AuctionHouseBot/AuctionHouseBotSeller.cpp

```diff
@@ -708,15 +708,15 @@ void AuctionBotSeller::SetPricesOfItem(ItemTemplate const* itemProto, SellerConf
 
     if (sAuctionBotConfig->GetConfig(CONFIG_AHBOT_BUYPRICE_SELLER))
         buyPrice = sellPrice;
+	
+	float basePriceFloat = (buyPrice * stackCount * priceRatio) / (itemProto->GetClass() == 6 ? 200.0f : static_cast<float>(itemProto->GetBuyCount())) / 100.0f;
+    float range = basePriceFloat * 0.04f;
+	
+    buyp = urand(static_cast<uint32>(basePriceFloat - range + 0.5f), static_cast<uint32>(basePriceFloat + range + 0.5f)) + 1;
```

Comment:
I'll make that change here.

Looking back, L717-719 may benefit from:
```cpp
basePriceFloat = buyp * 0.5f;
range = buyp * 0.4f;
bidp = static_cast<uint32>(frand(basePriceFloat - range, basePriceFloat + range) + 1.5f);
```

instead? Or is it not worth it since we've already lost precision due to `buyp` being uint32?